From caae473c1ce6f34597e91d609b87c2657b94ea70 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Thu, 28 Aug 2014 15:08:16 -0400
Subject: [iommu] Add empty stub for iommu_group_get_by_id()

Message-id: <20140828150816.13438.6920.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 89534
O-Subject: [PATCH RHEL7.1 BZ1123122] iommu: Add empty stub for iommu_group_get_by_id()
Bugzilla: 1123122
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1123122

Description:
------------
Almost every function in include/linux/iommu.h has an empty stub
but the iommu_group_get_by_id() did not get one by mistake.

This adds an empty stub for iommu_group_get_by_id() for IOMMU_API
disabled config.

Signed-off-by: Alexey Kardashevskiy <aik@ozlabs.ru>
Signed-off-by: Joerg Roedel <joro@8bytes.org>

RHEL Version Found:
-------------------
RHEL 7.0

Brew:
-----
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7892218

Upstream:
---------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=b62dfd29eeaf12f8bc79a50f680901e84b351851

Test Status:
------------
Currently being tested by IBM. This is part of their Power KVM product that we are bring into RHEL 7.1.

---------------------------------------------------------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 5fcb0d3..c5fe497 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -231,6 +231,11 @@ static inline struct iommu_domain *iommu_domain_alloc(struct bus_type *bus)
  return NULL;
 }
 
+static inline struct iommu_group *iommu_group_get_by_id(int id)
+{
+ return NULL;
+}
+
 static inline void iommu_domain_free(struct iommu_domain *domain)
 {
 }
-- 
1.7.1